src/meta/src/hummock/compaction/manual_compaction_picker.rs

```diff
@@ -167,6 +163,43 @@ impl ManualCompactionPicker {
             target_sub_level_id: 0,
         })
     }
+
+    /// Returns false if the given `sst` is rejected by filter defined by `option`.
+    /// Otherwise returns true.
+    fn filter_level_by_option(&self, level: &Level) -> bool {
+        let mut hint_sst_ids: HashSet<u64> = HashSet::new();
+        hint_sst_ids.extend(self.option.sst_ids.iter());
+        let tmp_sst_info = SstableInfo {
+            key_range: Some(self.option.key_range.clone()),
+            ..Default::default()
+        };
+        if self
+            .overlap_strategy
+            .check_overlap_with_tables(&[tmp_sst_info], &level.table_infos)
```

Comment:
Why return false? If this level is not overlap with given sst-id, we just skip this level rather than make this task failed.

Reply:
Oh, I made a mistake, it just continue. Ignore this.

Reply:
Return false means this specific level is not qualified as input level. The caller will skip this level, rather than fail the task.